Dockerfiles/Dockerfile

```diff
@@ -15,7 +12,7 @@ RUN if [ "${USE_LOCAL}" != "true" ]; then \
     fi
 
 ################################################################################
-FROM registry.access.redhat.com/ubi8/go-toolset:$GOLANG_VERSION as builder
+FROM registry.access.redhat.com/ubi8/go-toolset:latest as builder
```

Comment:
This might be a controversial part, and happy to revert it if there's consensus in that direction, but I think it's a good change for the following reasons:

- UBI go-toolset is a single-stream repo, so now that it has a 1.22 version, the 1.21 version is no longer maintained.
- We can rebuild the image to pick up on new Go fixes without having to necessarily update the `go` directive in the go.mod file.

I think these reasons are probably even more important further downstream where Konflux is already in use, since it will block releases if there are CVEs of a certain severity or higher.

Reply:
The only drawback I see would probably be to have to watch for the `go-toolset` to move to go 1.23 ?  (not sure if it would even happen and pretty sure quite long time form now)

Reply:
> The only drawback I see would probably be to have to watch for the `go-toolset` to move to go 1.23 ? (not sure if it would even happen and pretty sure quite long time form now)

That's too. I remember some conversation about latest, but do not remember details. It was decided not to use it. I'll try to find the thread.

Reply:
i agree with this. 
unless the images are released on major version, we should not use "latest" in general. 
also, to have a gap if downstream keeps with certain verson when ODH already moved to newer one, could cause backwards compability issue for the same code running between ODH and downstreaml

Reply:
> The only drawback I see would probably be to have to watch for the go-toolset to move to go 1.23 ?

@lburgazzoli Why would we need to watch for this though? It will be able to build older code, and we don't need to synchronously bump to 1.23 in go.mod at the same time: that can be done asynchronously at any point after (it just can't be done before). If it causes a build to break for some reason, we'll get notified of that ASAP and fix (and a temporary fix at that point might be to hold back to a previous minor version, but it's not a sustainable fix, because it's a single-stream image that we're basing on).

>  unless the images are released on major version, we should not use "latest" in general.

I don't understand this point, could you elaborate further?

> also, to have a gap if downstream keeps with certain verson when ODH already moved to newer one, could cause backwards compability issue for the same code running between ODH and downstream

Given that the UBI go-toolset image is single stream, and is used both upstream and downstream here, downstream will also want/need to keep updated to the latest version also. Any new downstream Z release even (think 2.8, 2.13, 2.16, etc) will eventually want to be built with the latest version of this image (meaning 1.22, where it's currently max 1.21).

By specifying the latest tag here, it means that Konflux/mintmaker will be able to take care of that for us automatically. Currently though, that's kinda "stuck" on each release, to the minor version, which means that manual intervention is needed to change that every time to a new minor tag. E.g. https://github.com/red-hat-data-services/rhods-operator/blob/main/Dockerfiles/Dockerfile.konflux#L4